.github/workflows/elixir-ci-v2.yml

```diff
@@ -0,0 +1,85 @@
+name: Elixir Quality CI
+
+
+on:
+  workflow_call:
+    inputs:
+          # -- Elixir version -- #
+          ci-container-version:
+            description: Which version of Elixir to use for the CI
+            default: elixir:1.8-otp-22-alpine
+            required: false
+            type: string
+          # -- Ignore steps -- #
+          ignore-credo:
+            description: ignore lint
+            default: true
+            required: false
+            type: boolean
+          # -- Services -- #
+          redis-version:
+            description: Redis version to use in tests. Ignored if empty
+            default: ""
+            required: false
+            type: string
+            
+permissions:
+  contents: read
+  
+jobs:
+  build:
+    name: Formatting, Credo, Sobelow, Deps audit, Tests
+    runs-on: self-hosted
+    env:
+      ImageOS: ubuntu22
+    container: ${{ inputs.ci-container-version }}
+    # Service containers to run with `container-job`
+    services:
+      # Label used to access the service container
+      redis:
+        # Docker Hub image
+        image: ${{ inputs.redis-version }}
+
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Restore dependencies cache
+        uses: actions/cache@v3
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+
+      - name: Install hex and rebar
+        run: "mix local.hex --force && mix local.rebar --force"
+
+      - name: Install dependencies
+        run: mix deps.get
+
+      - name: Check code formatting
+        run: mix format --check-formatted
+        # Check formatting even if there were unused deps so that
+        # we give devs as much feedback as possible & save some time.
+        if: always()
+
+      - name: Run Credo
+        run: mix credo --strict
+        # Run Credo even if formatting or the unused deps check failed
+        if: always()
+        continue-on-error: ${{ inputs.ignore-credo }}
+
+      - name: Check retired packages from hex
+        run: mix hex.audit
+        if: always()
+
+      - name: Run Sobelow
+        run: mix sobelow --skip
+        if: always()
```

Comment:
I don't expect libraries to have phoenix routers, thus there won't be a lot of errors returned by Sobelow

we can mark this task as making the ci fail

by defaut sobelow always exits with 0, you shall add the option "--exit" and add also "-i Config.HTTPS" to solve this only issue raised for redis_pool lib and that may happen in other libs

Reply:
the `if: always()` option is here to make steps run even if one higher fails
So basically when we create a pull request we will have all the checks and all the potential issues in one run, instead of fixing the format, committing, then fixing credo, committing, then ...

Okay, I will try without the -i Config.HTTPS just to see if we can use this job as CI for other projects too, if it fails I will add it

Reply:
Another option for the -i Config.HTTPS file would be to add a .sobelow-conf file with the -save-config option 🤔 
```elixir
[
  verbose: false,
  private: false,
  skip: true,
  router: "",
  exit: "low",
  format: "txt",
  out: "",
  threshold: "low",
  ignore: ["Config.HTTPS"],
  ignore_files: [""]
]
```